db/getItemsOnList_test.go

```diff
@@ -0,0 +1,104 @@
+package db
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+	"github.com/mount-joy/thelist-lambda/data"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type mockDB struct {
+	mock.Mock
+	dynamodbiface.DynamoDBAPI
+}
+
+func (m *mockDB) Query(input *dynamodb.QueryInput) (*dynamodb.QueryOutput, error) {
+	args := m.Called(input)
+	return args.Get(0).(*dynamodb.QueryOutput), args.Error(1)
+}
+
+func TestGetItemsOnList(t *testing.T) {
+	tests := []struct {
+		name        string
+		output      *dynamodb.QueryOutput
+		outputErr   error
+		expectedRes *[]data.Item
+		expectedErr error
+	}{
+		{
+			name:        "When the list is empty no items are returned",
+			output:      &dynamodb.QueryOutput{Items: []map[string]*dynamodb.AttributeValue{}},
+			outputErr:   nil,
+			expectedRes: &[]data.Item{},
+			expectedErr: nil,
+		},
+		{
+			name: "When the list contains items, they are returned",
+			output: &dynamodb.QueryOutput{
+				Items: []map[string]*dynamodb.AttributeValue{
+					{
+						"ListId": {S: aws.String("474c2Fff7")},
+						"Item":   {S: aws.String("Oranges")},
+						"Id":     {S: aws.String("1c2fa0a1")},
+					},
+					{
+						"ListId": {S: aws.String("474c2Fff7")},
+						"Item":   {S: aws.String("Apples")},
+						"Id":     {S: aws.String("bb0d5e8e")},
+					},
+				},
+			},
+			outputErr: nil,
+			expectedRes: &[]data.Item{
+				{
+					Item: "Oranges",
+					ID:   "1c2fa0a1",
+				},
+				{
+					Item: "Apples",
+					ID:   "bb0d5e8e",
+				},
+			},
+			expectedErr: nil,
+		},
+		{
+			name:        "When Query returns an error, that error is returned",
+			output:      nil,
```

Comment:
Would be good to have something valid returned from the mock here so that you are definitely testing that you don't return what was in the result if the err != nil